src/main/java/net/glowstone/entity/objects/GlowPainting.java

```diff
@@ -0,0 +1,334 @@
+        }
+
+        respawn();
+
+        return true;
+    }
+
+    @Override
+    public void setFacingDirection(BlockFace blockFace) {
+        setFacingDirection(blockFace, false);
+    }
+
+    @Override
+    public void pulse() {
+        super.pulse();
+
+        if (ticksLived % 11 == 0) {
+
+            if (location.getBlock().getRelative(getAttachedFace()).getType() == Material.AIR) {
+                if (EventFactory.callEvent(new HangingBreakEvent(this, RemoveCause.PHYSICS)).isCancelled()) {
+                    return;
+                }
+                world.dropItemNaturally(location, new ItemStack(Material.PAINTING));
+                remove();
+            }
+        }
+
+        if (ticksLived % (20 * 5) == 0 && isObstructed()) {
+            if (EventFactory.callEvent(new HangingBreakEvent(this, RemoveCause.PHYSICS)).isCancelled()) {
+                return;
+            }
+            if (location.getBlock().getRelative(getAttachedFace()).getType() == Material.AIR) {
+                world.dropItemNaturally(location, new ItemStack(Material.PAINTING));
+                remove();
+            }
+        }
+    }
+
+    @Override
+    protected void pulsePhysics() {
+        // not affected by physics
+    }
+
+    public boolean isObstructed() {
+        Location current = getTopLeftCorner();
+        BlockFace right = getRightFace();
+
+        for (int y = 0; y < art.getBlockHeight(); y++) {
+            for (int x = 0; x < art.getBlockWidth(); x++) {
+                if (!canHoldPainting(current)) {
+                    return true;
+                }
+                current = current.getBlock().getRelative(right).getLocation();
+            }
+
+            current = current.getBlock().getRelative(BlockFace.DOWN).getLocation();
+
+            // reset x and z
+            current.subtract(right.getModX() * art.getBlockWidth(), 0, right.getModZ() * art.getBlockWidth());
+        }
+
+        return !this.world.getEntityManager().getEntitiesInside(this.boundingBox, this).isEmpty();
+    }
+
+    private boolean canHoldPainting(Location where) {
+        Block block = where.getBlock();
+        if (block.getType().isSolid()) {
+            return false;
+        }
+
+        Block behind = block.getRelative(getAttachedFace());
+        return behind.getType().isSolid();
+    }
+
+    private Location getTopLeftCorner() {
+        BlockFace left = getLeftFace();
+        Location topLeft = center.clone();
+        int topMod = (int) getArtHeight();
+        int widthMod = (int) getArtWidth();
+
+        topLeft.add(left.getModX() * widthMod, topMod, left.getModZ() * widthMod);
+        return topLeft;
+    }
+
+    private BlockFace getLeftFace() {
+        return HangingFace.values()[(facing.ordinal() + 1) % 4].getBlockFace();
+    }
+
+    private BlockFace getRightFace() {
+        return getLeftFace().getOppositeFace();
+    }
+
+    @Override
+    public double getWidth() {
+        return 0.5;
```

Comment:
Add a note about this value being static (like you mentioned in the PR description)